internal/file/file_plugin.go

```diff
@@ -191,23 +191,41 @@ func (fp *FilePlugin) handleConfigApplyRequest(ctx context.Context, msg *bus.Mes
 			"instance_id", configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId(),
 			"error", err,
 		)
+
 		response = fp.createDataPlaneResponse(
 			correlationID,
 			mpi.CommandResponse_COMMAND_STATUS_ERROR,
 			"Config apply failed, rolling back config",
 			configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId(),
 			err.Error(),
 		)
+		fp.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: response})
+
+		rollbackErr := fp.fileManagerService.Rollback(
+			ctx,
+			configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId(),
+		)
+		if rollbackErr != nil {
+			rollbackResponse := fp.createDataPlaneResponse(
+				correlationID,
+				mpi.CommandResponse_COMMAND_STATUS_FAILURE,
```

Comment:
Could we add a check in the unit test to check the second messages status and message ?

Reply:
Added